### PR TITLE
Add single pokemon search form, API, display component, and draft err…

### PR DESCRIPTION
**Pull request by CaseyConlin**
There's a separate API service for a single Pokemon along with a form to feed the search and a crude component to display the result. The API service and App do provide some basic error handling.

### CODE REVIEW

File: src/services/single-pokemon.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export const getSinglePokemon = async (pokemon) => {
+  const response = await fetch(
+    "https://pokeapi.co/api/v2/pokemon/" + pokemon.toLowerCase()
+  );
+  if (response.status === 404) {
+    throw Error(
+      "Shoot, that Pokemon's not in the Pokedex, please enter a valid Pokemon."
```
Comment (ifyoumakeit) [inline src/services/single-pokemon.js:7]: **praise:** ‏Ha! Love this!

File: src/services/single-pokemon.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    );
+  }
+  const data = await response.json();
+
+  const singlePokemon = [
+    { name: data.name, id: data.id, image: data.sprites.front_default },
+  ];
+  return singlePokemon;
```
Comment (ifyoumakeit) [inline src/services/single-pokemon.js]:
```suggestion
  return { name: data.name, id: data.id, image: data.sprites.front_default };
```


**suggestion:** ‏A `singlePokemon` suggests an object to me, not an array of one. I would simplify this function to only return a single data point. That way your component doesn't need to pull off the first item of the array.

File: src/components/single-pokemon-component.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    <div>
+      <p>{props.singlePokemonData[0].name}</p>
+      <p>{props.singlePokemonData[0].id}</p>
+      <img
+        src={props.singlePokemonData[0].image}
+        alt={props.singlePokemonData[0].name}
+      />
+    </div>
+  );
+};
```
Comment (ifyoumakeit) [inline src/components/single-pokemon-component.js]: re: https://github.com/Parkcrest/Pokedex/pull/8/files#r1068293096

```suggestion
export const SinglePokemonComponent = ({name, id, image}) => {
  return (
    <div>
      <p>{name}</p>
      <p>{id}</p>
      <img
        src={image}
        alt={name}
      />
    </div>
  );
};
```

File: src/app.tsx [hunk 4]
```diff
@@ -19,9 +25,36 @@ export const App = () => {
...
     <div>
       <div className="App">
+        <SinglePokemonForm
+          value={searchValue}
+          change={searchValueChangeHandler}
+          click={singlePokemonClickHandler}
+        />
+
+        {singlePokemon ? (
+          <SinglePokemonComponent singlePokemonData={singlePokemon} />
```
Comment (ifyoumakeit) [inline src/app.tsx]: re: https://github.com/Parkcrest/Pokedex/pull/8/files#r1068293096

```suggestion
          <SinglePokemonComponent {...singlePokemon} />
```

File: src/app.tsx [hunk 5]
```diff
@@ -15,7 +15,11 @@ import { SinglePokemonForm } from "./components/single-pokemon-form";
 export const App = () => {
   const [items, setItems] = useState<Pokemon[]>([]);
   const [searchValue, setSearchValue] = useState("Pikachu");
-  const [singlePokemon, setSinglePokemon] = useState([{}]);
+  const [singlePokemon, setSinglePokemon] = useState({
+    name: "",
+    id: "",
+    image: "",
+  });
```
Comment (CaseyConlin) [inline src/app.tsx:22]: There may be a cleaner way to do this, but the refactoring required including assignments for the singlePokemon properties to make the component work. These empty assignments were also added to line 38 of the singlePokemonClickHandler where the results get cleared.

**Pull request merged**